conference_scheduler/scheduler.py

```diff
@@ -3,23 +3,33 @@
 from conference_scheduler.resources import ScheduledItem
 
 
-def is_valid_schedule(schedule):
-    if len(schedule) == 0:
+def _all_constraints(shape, sessions, events, X, constraints=None):
+    session_array = params.session_array(sessions)
+    tag_array = params.tag_array(events)
+    generators = [params.constraints(shape, session_array, tag_array, X)]
+    if constraints is not None:
+        generators.append(constraints)
+    for generator in generators:
+        for constraint in generator:
+            yield constraint
+
+
+def is_valid_solution(solution, shape, sessions, events, constraints=None):
+    if len(solution) == 0:
         return False
-    return True
+    return all([c for c in _all_constraints(
```

Comment:
whoops! Yep. That's a leftover from about 4 different ways to do this. I'll sort it....